File: handlerclaw/lib/features/devices/domain/entities/user_device_entity.dart
class UserDeviceEntity {
  final String id;
  final String userId;
  final String deviceId;
  final String? deviceBrand;
  final String? deviceModel;
  final String? osVersion;
  final String fcmToken;
  final String platform;
  final String status;
  final DateTime? lastSeenAt;
  final DateTime createdAt;
  final DateTime updatedAt;

  UserDeviceEntity({
    required this.id,
    required this.userId,
    required this.deviceId,
    this.deviceBrand,
    this.deviceModel,
    this.osVersion,
    required this.fcmToken,
    required this.platform,
    required this.status,
    this.lastSeenAt,
    required this.createdAt,
    required this.updatedAt,
  });

  bool get isOnline => status == 'ACTIVE';
}

File: api/src/modules/user-device/domain/entities/user-device.entity.ts
import type {
  UserDevicePlatform,
  UserDeviceStatus,
} from "../../../../lib/generated/prisma/enums.js";

export type UserDevice = {
  id: string;
  user_id: string;
  status: UserDeviceStatus;
  device_id: string;
  device_brand: string | null;
  device_model: string | null;
  os_version: string | null;
  fcm_token: string;
  platform: UserDevicePlatform;
  last_seen_at: Date | null;
  created_at: Date;
  updated_at: Date;
};
